### PR TITLE
docs: improve join docs

### PR DESCRIPTION
**Pull request by mjsax**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/joins/join-streams-and-tables.md [hunk 1]
```diff
@@ -40,6 +40,9 @@ CREATE STREAM pageviews_enriched AS
   EMIT CHANGES;
 ```
 
+> Note: Currently, the join expression must be a simple comparison.
```
Comment (JimGalasyn) [inline docs/developer-guide/joins/join-streams-and-tables.md:43]:
```suggestion
!!! note
    The join expression must be a simple comparison. Non-equi joins,
```

File: docs/developer-guide/joins/join-streams-and-tables.md [hunk 2]
```diff
@@ -40,6 +40,9 @@ CREATE STREAM pageviews_enriched AS
...
+Non-equi joins, or joins on multiple columns are not supported yet.
```
Comment (JimGalasyn) [inline docs/developer-guide/joins/join-streams-and-tables.md:44]:
```suggestion
    or joins on multiple columns are not supported. 
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a suggestion for note styling.

File: docs/developer-guide/joins/join-streams-and-tables.md (hunk 1 above)
Comment (mjsax) [reply docs/developer-guide/joins/join-streams-and-tables.md:43]: We use `>` for "note" boxes below, too.
```
>Note: Only stream-stream joins are windowed.
```

Should we update all of them?

Comment (JimGalasyn) [reply docs/developer-guide/joins/join-streams-and-tables.md:43]: @mjsax If you feel like it, but it's not a big deal.

File: docs/developer-guide/joins/join-streams-and-tables.md (hunk 2 above)
Comment (blueedgenick) [reply docs/developer-guide/joins/join-streams-and-tables.md:44]: maybe worth adding a few words to say that if you want multti-col joins your workaround is to `join` on key equality and add a where clause to compare the other relevant cols ?

File: docs/developer-guide/joins/join-streams-and-tables.md (hunk 1 above)
Comment (blueedgenick) [reply docs/developer-guide/joins/join-streams-and-tables.md:43]: i think even this " must be a simple comparison" could be tightened up a little - to me a "simple comparison" is not quite the same as "an equality check" - could maybe include things like !=, >=, etc

Comment (mjsax) [reply docs/developer-guide/joins/join-streams-and-tables.md:43]: Doesn't the second sentence not cover it: `Non-equi joins ... are not supported.` ?

Comment (blueedgenick) [reply docs/developer-guide/joins/join-streams-and-tables.md:43]: absolutely it does! just suggesting that the first sentence could also be more specific. not a big deal.

**Comment by mjsax** [comment]
Finally updated this PR. If there are no objections, will merge after Jenkins passed.

@JimGalasyn Would be great to get this live soon. Can you help to cherry-pick?

**Comment by JimGalasyn** [comment]
@mjsax Will do!

**Pull request merged**